modules/ipp/scanner.go

```diff
@@ -555,14 +674,24 @@ func (scanner *Scanner) newIPPScan(target *zgrab2.ScanTarget) *scan {
 	return &newScan
 }
 
+// Cleanup closes any connections that have been opened during the scan
+func (scan *scan) Cleanup() {
+	if scan.connections != nil {
+		for _, conn := range scan.connections {
+			defer conn.Close()
+		}
+		scan.connections = nil
+	}
+}
+
 // TODO: Do you want to retry with TLS for all versions? Just one's you've already tried? Haven't tried? Just the same version?
-func (scanner *Scanner) tryGrabForVersions(target *zgrab2.ScanTarget, versions *[]version) (*scan, *zgrab2.ScanError) {
-	scan := scanner.newIPPScan(target)
-	// TODO: Implement scan.Cleanup()
+func (scanner *Scanner) tryGrabForVersions(target *zgrab2.ScanTarget, versions []version, tls bool) (*scan, *zgrab2.ScanError) {
+	scan := scanner.newIPPScan(target, tls)
+	defer scan.Cleanup()
 	var err *zgrab2.ScanError
-	for i := 0; i < len(*versions); i++ {
-		err = scanner.Grab(scan, target, &(*versions)[i])
-		if err != nil && err.Err == ErrVersionNotSupported && i < len(*versions)-1 {
+	for i := 0; i < len(versions); i++ {
+		err = scanner.Grab(scan, target, &(versions)[i])
```

Comment:
The leftover parens around `versions` make it look like it's trying to do something weird with the order of the index/address operations.